### PR TITLE
test: remove hollow never-wired-mock tests from test_performance

### LINKED ISSUE #897: test: test_large_pagination_performance has a disconnected mock (never wired)
## Problem

`tests/test_performance.py::test_large_pagination_performance` builds `mock_responses` (10 pages) and `mock_api_method`, but **never wires them** — it calls `get_all_products.asyncio_detailed(client=katana_client, limit=250)` without patching that method or `katana_client`'s transport. So the test exercises the real transport stack and its `except Exception` block swallows any network error, meaning it can pass without asserting anything meaningful.

Surfaced during the fake-time sweep (#894), which removed the test's only "assertion" (`assert duration < 2.0` — a flaky wall-clock bound that was actually measuring network latency, not pagination).

## Options
- Wire the mock properly (patch the inner transport / `get_all_products`) so it actually verifies 10-page auto-pagination, **or**
- Convert to a clearly-labeled live integration test gated on `KATANA_TEST_API_KEY`, **or**
- Delete it if `test_transport_auto_pagination.py` already covers pagination (the test's own comment says it does).

## Context
Pre-existing; orthogonal to the #894 time-faking fix, so split out here rather than scope-creeping that PR.

### PR DESCRIPTION
**Pull request by dougborg**
## Summary

Closes #897.

`test_large_pagination_performance` built 10 mock paginated responses but **never wired them in** — it then called `get_all_products` against a real (fake-cred) `katana_client` and swallowed the resulting network error, asserting nothing about pagination. Its only ever-real assertion (a wall-clock `duration < 2.0`) was removed in #894 as a flaky timing bound.

Sweeping the file surfaced **three more tests with the same never-wired / tautological pattern**:
- `test_large_result_set_memory` — identical hollow pattern; doesn't measure memory.
- `test_retry_on_server_error` — defines an unused `failing_method`, then calls a *local* mock and asserts `result is not None`. Tests no client retry.
- `test_no_retry_on_client_error` — calls a local mock once, asserts `call_count == 1`. Tautological.

All four are deleted. The behavior they *pretended* to cover is covered properly elsewhere:
- **Multi-page auto-pagination** → `tests/test_transport_auto_pagination.py` (real `MockTransport`, asserts combined `data` across pages, plus stop-on-error and limit edge cases).
- **Retry behavior** → `tests/test_rate_limit_retry.py` and `tests/test_rate_limit_transport.py`.

Kept the genuinely-wired tests, reorganized into `TestConcurrency` (`test_concurrent_requests`, `test_concurrent_pagination` — the latter correctly `patch.object`-wires the API seam) and `TestClientLifecycle` (`test_client_cleanup`). Dropped the now-unused `AsyncMock` import.

Net: **+38 / −231** — removing dead, misleading tests, not coverage. Verified the full suite + coverage report still green (`3885 passed`).

## Test plan
- [x] `uv run poe test-coverage` → 3885 passed, no regression
- [x] `uv run ruff check` / `uv run pyright` clean on the file
- [x] `test:` type — no release impact

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test_performance.py [hunk 1]
```diff
@@ -100,173 +57,17 @@ async def make_mock(result_id: int):
...
-    """Test concurrent usage and potential race conditions."""
-
     @pytest.mark.asyncio
     @pytest.mark.slow
     async def test_concurrent_pagination(
         self, katana_client, mock_paginated_response_factory
     ):
-        """Test multiple concurrent pagination operations."""
+        """Multiple concurrent paginated calls each return a response without
+        cross-talk or unhandled exceptions."""
```
[the remark this answers is not in the document]
Comment (dougborg) [reply tests/test_performance.py]: Fixed in 2c5cc34b — you're right, it never paginated (patched `asyncio_detailed`, only returned each set's first page). Rather than just soften the docstring, I made the test honest about what it actually verifies: renamed it `test_concurrent_calls_return_independent_results`, dropped the unused multi-page machinery, and it now asserts each distinct response object comes back exactly once (by identity) — i.e. concurrent dispatch with no cross-talk. Pagination stays covered in `test_transport_auto_pagination.py`.
